java/src/jmri/jmrit/signalling/SignallingSourcePanel.java

```diff
@@ -180,7 +169,10 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
             boolean newValue = (Boolean) e.getNewValue();
             discoverPairs.setEnabled(newValue);
         }
-        //log.debug("SSP 179 Event: {}", e.getPropertyName());
+        log.debug("SSP 173 Event: {}; Source: {}", e.getPropertyName(), e.toString()); // doesn't get notified, newDestination
+        if (e.getPropertyName().equals("Frame.active")) { // a blunt way of redrawing the Pairs table
```

Comment:
@silverailscolo You probably want to listen to changes in the master source of signal mast pairs. If that source does not fire a PropertyChangeEvent or IndexedPropertyChangeEvent when a pair is added, it should be modified to fire one of those events, so you can listen to it.